src/shared/extensionUtilities.ts

```diff
@@ -69,3 +73,53 @@ export function safeGet<O, T>(obj: O | undefined, getFn: (x: O) => T): T | undef
 
     return undefined
 }
+
+/**
+ * Helper function to create a webview containing the welcome page
+ * Returns an unfocused vscode.WebviewPanel if the welcome page is renderable.
+ * Returns void if the welcomePage.html file is unrenderable
+ *
+ * @param context VS Code Extension Context
+ * @param page Page to load (use for testing); default: `welcomePage.html`
+ */
+export async function createWelcomeWebview(
+    context: vscode.ExtensionContext,
+    page: string = 'welcomePage.html'
+): Promise<vscode.WebviewPanel | void> {
+    let html: string | undefined
+    try {
+        html = convertExtensionRootTokensToPath(
+            await readFileAsString(path.join(context.extensionPath, page)),
+            context.extensionPath
+        )
+        if (!html) {
+            throw new Error()
```

Comment:
1. Is anything logged indicating that the path could not be found (or any other error that may occur)?

2. A command will trigger this, usually through explicit request from the user. We should put a message in the output channel so they know nothing is going to happen. Outputting a message to the user is not the responsibility of this method, perhaps the method should not eat the error, and the command code that calls this method can catch+message

Reply:
No, what would be the most ideal place to surface this? A toast? Right now, the welcome page will only show on a user directly invoking this command, so immediate feedback might be best.

Reply:
Also, this was largely done to mitigate https://github.com/aws/aws-toolkit-vscode/pull/619#discussion_r294975639

Reply:
I think the Output Channel would work. The reason we wouldn't place the notification directly in this method is that this method could be called from elsewhere, and it may not make sense in those contexts to surface a similar message. 

Re: The comment you're trying to mitigate, there appears to still be logic in the async handler. Could all of that be placed in its own method?

Reply:
re: second point, yes, I can add that logic to `createWelcomeWebview`. In that case, I'll swap the name to `showWelcomeWebview` since that's what it'll be doing (or having a `showWelcomeWebview` that wraps `create` and does the explicit focus).

re: the first point, this is an error state that a user should never see, and if the user does see it, it would be in direct reaction to a user action (in its current state). I'm not sure if the output channel is enough feedback in a case like this...as a user, if I click something, I expect some sort of immediate, obvious visual feedback. I don't feel that an output channel that may already contain information is the best way to surface this type of failure. I can't think of any scenario that we'd call for the welcome page with absolutely no user input.

That being said, we should send any failures with this as a telemetry event. I'll look into that.